internal/log_analysis/rules_engine/src/rule.py

```
@@ -211,57 +274,139 @@ def _get_dedup(self, event: Mapping, title: Optional[str], use_default_on_except
-                return None
-            raise
+            self.logger.warning('description method for rule with id [%s] raised exception. Using default Exception: %s', self.rule_id, err)
+            return None
 
-        if len(title_string) > MAX_TITLE_SIZE:
-            # If title exceeds max size, truncate it
+        if len(description) > MAX_CUSTOM_FIELD_SIZE:
+            # If custom field exceeds max size, truncate it
             self.logger.warning(
-                'maximum title string size is [%d] characters. Title for rule with ID '
-                '[%s] is [%d] characters. Truncating.', MAX_TITLE_SIZE, self.rule_id, len(title_string)
+                'maximum field [description] length is [%d]. [%d] for rule with ID [%s] . Truncating.',
+                MAX_CUSTOM_FIELD_SIZE, len(description), self.rule_id,
             )
-            num_characters_to_keep = MAX_TITLE_SIZE - len(TRUNCATED_STRING_SUFFIX)
-            return title_string[:num_characters_to_keep] + TRUNCATED_STRING_SUFFIX
+            num_characters_to_keep = MAX_CUSTOM_FIELD_SIZE - len(TRUNCATED_STRING_SUFFIX)
+            return description[:num_characters_to_keep] + TRUNCATED_STRING_SUFFIX
+        return description
 
-        return title_string
+    def _get_destination_override(self, event: Mapping) -> Optional[List[str]]:
+        if not hasattr(self._module, 'destination_override'):
+            return None
 
-    def _get_alert_context(self, event: Mapping, use_default_on_exception: bool = True) -> Optional[str]:
-        if not self._has_alert_context:
+        try:
+            command = getattr(self._module, 'destination_override')
+            destination_override = self._run_command(command, event, list())
+        except Exception as err:  # pylint: disable=broad-except
+            self.logger.warning('_get_destination_override method raised exception. Exception: %s', err)
```

The previous code had a variable passed to all methods: `use_default_on_exception`. 
The purpose of this exception: In case of normal operation, if any of the methods that generate the new fields fail, we just want to use the default. 
If any of the methods fail while user is testing through the UI, we want to return the exception.

---

Gotcha, will change the `_get_field` functions to accept the `use_default_on_exception` and raise exceptions instead of logging a warning when `use_default_on_exception` is false.